### PR TITLE
Add DB configurations

### PR DESCRIPTION
**Pull request by ariefrahmansyah**
**Which issue(s) this PR fixes**:

Fixes broken pipe from Merlin to DB

**Does this PR introduce a user-facing change?**:

```release-note

```

**Checklist**

- [ ] Added unit test, integration, and/or e2e tests
- [ ] Tested locally
- [ ] Updated documentation
- [ ] Update Swagger spec if the PR introduce API changes
- [ ] Regenerated Golang and Python client if the PR introduce API changes

### CODE REVIEW

File: api/config/config.go [hunk 1]
```diff
@@ -124,6 +124,10 @@ type DatabaseConfig struct {
 	Password      string `envconfig:"DATABASE_PASSWORD" required:"true"`
 	Database      string `envconfig:"DATABASE_NAME" default:"mlp"`
 	MigrationPath string `envconfig:"DATABASE_MIGRATIONS_PATH" default:"file://db-migrations"`
+
+	ConnMaxIdleTime time.Duration `envconfig:"DATABASE_CONN_MAX_IDLE_TIME" default:"10m"`
+	MaxIdleConns    int           `envconfig:"DATABASE_MAX_IDLE_CONNS" default:"10"`
```
Comment (pradithya) [inline api/config/config.go:129]: Can we make it 0 as default?

File: api/config/config.go [hunk 2]
```diff
@@ -124,6 +124,10 @@ type DatabaseConfig struct {
...
+	MaxOpenConns    int           `envconfig:"DATABASE_MAX_OPEN_CONNS" default:"100"`
```
Comment (pradithya) [inline api/config/config.go:130]: Do we need to update helm chart to include these?

File: api/cmd/api/setup.go [hunk 3]
```diff
@@ -56,6 +56,16 @@ func initDB(cfg config.DatabaseConfig) (*gorm.DB, func()) {
...
 	db.LogMode(false)
+
+	sqlDB := db.DB()
+	if sqlDB == nil {
+		panic("fail to get the underlying database connection")
+	}
+
+	sqlDB.SetConnMaxIdleTime(cfg.ConnMaxIdleTime)
+	sqlDB.SetMaxIdleConns(cfg.MaxIdleConns)
+	sqlDB.SetMaxOpenConns(cfg.MaxOpenConns)
```
Comment (pradithya) [inline api/cmd/api/setup.go:67]: Can you add `SetConnMaxLifetime` (https://github.com/go-sql-driver/mysql/issues/529) ?

File: api/config/config.go (hunk 2 above)
Comment (ariefrahmansyah) [reply api/config/config.go]: Updated

File: api/config/config.go (hunk 1 above)
Comment (ariefrahmansyah) [reply api/config/config.go]: Updated

File: api/cmd/api/setup.go (hunk 3 above)
Comment (ariefrahmansyah) [reply api/cmd/api/setup.go:68]: Added

File: charts/merlin/values.yaml [hunk 4]
```diff
@@ -245,6 +246,11 @@ merlin-postgresql:
   # The password secret will be created using this password
   postgresqlPassword: merlin
 
+  connMaxIdleTime: 0
```
Comment (pradithya) [inline charts/merlin/values.yaml:249]: I think it should be string and non 0
```
connMaxIdleTime: 10m
connMaxLifetime: 30m
```

**Review by pradithya — APPROVED** [review]
LGTM!!  thanks @ariefrahmansyah

**Review by karzuo — APPROVED** [review]
LGTM

**Pull request merged**